lib/util/stream_utils.js

```diff
@@ -425,7 +425,44 @@ shaka.util.StreamUtils = class {
     await shaka.util.StreamUtils.getDecodingInfosForVariants(
         manifest.variants, usePersistentLicenses, /* srcEquals= */ false);
     manifest.variants = manifest.variants.filter((variant) => {
+      // See: https://github.com/google/shaka-player/issues/3860
       const video = variant.video;
+      const ContentType = shaka.util.ManifestParserUtils.ContentType;
+      if (video) {
+        let videoCodecs = video.codecs;
+        // For multiplexed streams.
+        if (video.codecs.includes(',')) {
+          const allCodecs = video.codecs.split(',');
+          videoCodecs = shaka.util.ManifestParserUtils.guessCodecs(
+              ContentType.VIDEO, allCodecs);
+          videoCodecs = shaka.util.StreamUtils.patchVp9(videoCodecs);
+          const audioCodecs = shaka.util.ManifestParserUtils.guessCodecs(
+              ContentType.AUDIO, allCodecs);
+          const audioFullType = shaka.util.MimeUtils.getFullOrConvertedType(
+              video.mimeType, audioCodecs, ContentType.AUDIO);
+          if (!MediaSource.isTypeSupported(audioFullType)) {
+            return false;
+          }
+        }
+        videoCodecs = shaka.util.StreamUtils.patchVp9(videoCodecs);
+        const fullType = shaka.util.MimeUtils.getFullOrConvertedType(
+            video.mimeType, videoCodecs, ContentType.VIDEO);
+        if (!MediaSource.isTypeSupported(fullType)) {
+          return false;
+        }
+      }
+      const audio = variant.audio;
+      if (audio) {
+        const codecs =
+            (audio.codecs.toLowerCase() == 'ac-3' &&
+            shaka.util.Platform.isTizen()) ? 'ec-3' : audio.codecs;
```

Comment:
I assume this Tizen-specific logic must already exist in another location, so I think it would be best to encapsulate it in a method somewhere.

Reply:
Done!